plugin/pkg/scheduler/algorithm/predicates/utils_test.go

```diff
@@ -68,3 +71,116 @@ func ExampleFindLabelsInSet() {
 	// label1=value1,label2=value2,label3=will_see_this
 	// pod1,pod2,
 }
+
+func TestFindLabelsInSet(t *testing.T) {
+
+	nsPod := v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pod1",
+			Namespace: "ns1",
+			Labels: map[string]string{
+				"label1": "foo",
+				"label2": "foo",
+				"label3": "bar",
+			},
+		},
+	}
+
+	expected := map[string]string{
+		"label1": "foo",
+		"label2": "foo",
+	}
+
+	result := FindLabelsInSet([]string{"label1", "label2"}, nsPod.ObjectMeta.Labels)
+	if !reflect.DeepEqual(result, expected) {
+		t.Errorf("Got different result than expected.\nDifference detected on:\n%s", diff.ObjectReflectDiff(expected, result))
+	}
+
+}
+
+func TestAddUnsetLabelsToMap(t *testing.T) {
+
+	labelSubset := labels.Set{}
+	labelSubset["label1"] = "value1"
+	labelSubset["label2"] = "value2"
+
+	nsPod := v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pod1",
+			Namespace: "ns1",
+			Labels: map[string]string{
+				"label1": "foo",
+				"label2": "foo",
+				"label3": "bar",
+			},
+		},
+	}
+
+	expected := labels.Set{
+		"label1": "value1",
+		"label2": "value2",
+		"label3": "bar",
+	}
+
+	AddUnsetLabelsToMap(labelSubset, []string{"label1", "label2", "label3"}, nsPod.ObjectMeta.Labels)
+
+	if !reflect.DeepEqual(labelSubset, expected) {
+		t.Errorf("Got different result than expected.\nDifference detected on:\n%s", diff.ObjectReflectDiff(expected, labelSubset))
+	}
+
+}
+
+func TestFilterPodsByNamespace(t *testing.T) {
+
+	nsPods := []*v1.Pod{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pod1",
+				Namespace: "ns1",
+				Labels: map[string]string{
+					"label1": "foo",
+					"label2": "foo",
+					"label3": "bar",
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pod2",
+				Namespace: "ns1",
+			},
+		},
+
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "pod3ThatWeWontSee",
+			},
+		},
+	}
+
+	expected := []*v1.Pod{
```

Comment:
Please add the entries of `nsPods` to `expected` instead of repeating Pod definitions.